Source/ACE/Network/GameAction/Actions/GameActionQueryHealth.cs

```diff
@@ -0,0 +1,41 @@
+﻿using ACE.Entity;
+using ACE.Managers;
+using ACE.Network.GameEvent.Events;
+using System;
+
+namespace ACE.Network.GameAction.Actions
+{
+    [GameAction(GameActionType.QueryHealth)]
+    public class GameActionQueryHealth : GameActionPacket
+    {
+        private ObjectGuid objectId;
+
+        public GameActionQueryHealth(Session session, ClientPacketFragment fragment) : base(session, fragment) { }
+
+        public override void Read()
+        {
+            uint fullId = Fragment.Payload.ReadUInt32();
+            this.objectId = new ObjectGuid(fullId);
+        }
+
+        public override void Handle()
+        {
+            // TODO: Extend to creatures as well, as we momentarily only have players
+
+            if (objectId.IsPlayer())
+            {
+                var tmpLandblock = this.Session.Player.Position.LandblockId;
+                var pl = (Player)LandblockManager.GetObjectByGuid(tmpLandblock, objectId);

```

Comment:
you already have the player.... this.Session.Player

There should be next to zero reason to return objects from the Landblock / LandblockManager.  That was a red flag to me, this is incorrect usage.  It's also not safe to extract as the player may cross a landblock boundary while running across the open world.